File: app/src/main/java/space/pal/sig/util/SharedPreferencesUtil.java
package space.pal.sig.util;

import android.content.Context;
import android.content.SharedPreferences;

import androidx.preference.PreferenceManager;

import com.google.gson.Gson;
import com.google.gson.JsonSyntaxException;

import java.util.HashSet;
import java.util.Set;

/**
 * SpaceNow
 * Created by Catalin on 7/22/2020
 **/
public class SharedPreferencesUtil {

    private static final String SPACE_PREFERENCES = "Space.Shared.Preferences";
    private final SharedPreferences preferences;
    private final Gson gson;

    public SharedPreferencesUtil(Context context, Gson gson) {
        this.preferences = PreferenceManager.getDefaultSharedPreferences(context);
        this.gson = gson;
    }

    public void storeInteger(String key, Integer value) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putInt(key, value);
        editor.apply();
    }

    public void storeBoolean(String key, Boolean value) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putBoolean(key, value);
        editor.apply();
    }

    public void storeLong(String key, Long value) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putLong(key, value);
        editor.apply();
    }

    public void storeFloat(String key, Float value) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putFloat(key, value);
        editor.apply();
    }

    public void storeString(String key, String value) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putString(key, value);
        editor.apply();
    }

    public void storeSet(String key, Set<String> items) {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putStringSet(key, items);
        editor.apply();
    }

    public void storeObject(String key, Object object) {
        storeString(key, gson.toJson(object));
    }

    public Integer getInteger(String key, Integer defaultValue) {
        return preferences.getInt(key, defaultValue);
    }

    public Boolean getBoolean(String key, Boolean defaultValue) {
        return preferences.getBoolean(key, defaultValue);
    }

    public Long getLong(String key, Long defaultValue) {
        return preferences.getLong(key, defaultValue);
    }

    public Float getFloat(String key, Float defaultValue) {
        return preferences.getFloat(key, defaultValue);
    }

    public String getString(String key, String defaultValue) {
        return preferences.getString(key, defaultValue);
    }

    public <T> T getObject(String key, Class<T> cls) {
        try {
            return gson.fromJson(getString(key, null), cls);
        } catch (JsonSyntaxException ex) {
            ex.printStackTrace();
            return null;
        }
    }

    public Set<String> getSet(String key) {
        return preferences.getStringSet(key, new HashSet<>());
    }

}

File: app/src/main/java/space/pal/sig/util/SharedPreferencesUtil.kt
package space.pal.sig.util

import android.app.Application
import android.content.Context
import android.content.SharedPreferences

/**
SpaceNow
Created by Catalin on 12/6/2020
 **/
class SharedPreferencesUtil(
        context: Application
) {

    companion object {
        private const val SPACE_NOW_PREF = "space_now_pref"
    }

    private val sharedPreferences = context.getSharedPreferences(
            SPACE_NOW_PREF, Context.MODE_PRIVATE)

    fun save(key: String, value: String) {
        sharedPreferences.edit().putString(key, value).apply()
    }

    fun get(key: String, defaultValue: String?): String? {
        return sharedPreferences.getString(key, defaultValue)
    }

    fun save(key: String, value: Int) {
        sharedPreferences.edit().putInt(key, value).apply()
    }

    fun get(key: String, defaultValue: Int): Int {
        return sharedPreferences.getInt(key, defaultValue)
    }

    fun save(key: String, value: Long) {
        sharedPreferences.edit().putLong(key, value).apply()
    }

    fun get(key: String, defaultValue: Long): Long {
        return sharedPreferences.getLong(key, defaultValue)
    }

    fun save(key: String, value: Boolean) {
        sharedPreferences.edit().putBoolean(key, value).apply()
    }

    fun get(key: String, defaultValue: Boolean): Boolean {
        return sharedPreferences.getBoolean(key, defaultValue)
    }

    fun save(key: String, items: Set<String>) {
        val editor: SharedPreferences.Editor = sharedPreferences.edit()
        editor.putStringSet(key, items)
        editor.apply()
    }

    fun get(key: String): MutableSet<String> {
        return sharedPreferences.getStringSet(key, mutableSetOf())!!
    }

    fun remove(key: String) {
        sharedPreferences.edit().remove(key).apply()
    }

}